### PR TITLE
[WiP] Canary deployments with flagger

### PR DESCRIPTION
**Pull request by carlossg**

### CODE REVIEW

**Comment by carlossg** [comment]
@jstrachan is this what you had in mind ?

File: packs/go/charts/templates/canary.yaml [hunk 1]
```diff
@@ -0,0 +1,44 @@
+{{- if eq .Release.Namespace "jx-production" }}
```
Comment (ccojocar) [inline packs/go/charts/templates/canary.yaml]: Shouldn't we make this configurable over a value?

**Comment by polothy** [comment]
First, this is really awesome, really appreciate the work on this!

There's however some pretty specific information in the values, like namespace, hosts and gateways.  Is there a way to set these from the JX environment?  Suppose we could always override the values since this is Helm, but you have to do that for every app in the environment.

I'm not coming up with any good ideas, but my point is, these values just seem pretty specific to the cluster and environment.  Thinking way down the road where you might deploy this app into multiple clusters.  Just wondering if there's some way to JX'ify them so they make sense for the cluster and environment that you are deploying to.

```
canary:
   enable: false
   namespace: jx-production
   service:
     hosts:
     - app.example.com
     gateways:
     - jx-gateway.istio-system.svc.cluster.local
```

**Comment by jstrachan** [comment]
I think another version of this PR got merged already so lets close this...

**Pull request closed without merging**